keras/backend/tensorflow_backend.py

```diff
@@ -3028,6 +3028,11 @@ def sparse_categorical_crossentropy(target, output, from_logits=False):
     # Returns
         Output tensor.
     """
+    # If the channels are in axis 1, move them to be the last axis:
+    if image_data_format() == 'channels_first':
```

Comment:
You cannot rely on `K.image_data_format` here.

The way the `data_format` argument works, everywhere is:
- it can be explicitly passed by the user
- if not passed (None) it defaults to `K.image_data_format`

We never force the use of `K.image_data_format`.

For cross_entropy loss functions, the proper approach would be to have an `axis` argument specifying the softmax axis (default `axis=-1`).

Reply:
@fchollet  If I simply add an `axis` argument with default `-1`, then the whole flow needs to pass the `axis`, or it will simply default to `-1` and will continue to fail when the `image_data_format` is `'channels_first'`. There is currently nowhere in the flow that checks or passes the axis or the data format. 

I could do what I see many other functions in `tensorflow_backend` doing, and pass an argument `data_format` with default `data_format=None`. Then inside the function I could do this check:
```python
if data_format is None:
    data_format = image_data_format()
if data_format not in {'channels_first', 'channels_last'}:
    raise ValueError('Unknown data_format: ' + str(data_format))
```
That would satisfy the requirements you wrote (that the data format can be explicitly passed by the user, and if not passed will default to `K.image_data_format`), while solving the bug we see with `'channels_first'` data format. It is also consistent with the behavior of other functions in `tensorflow_backend`.